### ISSUE TITLE
All messages should be declared in Check 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#13189]
Detected at https://github.com/checkstyle/checkstyle/pull/13129#issuecomment-1582698277

we need to do refactoring to of modules to move all messages MSG_(currently) fields and log() to final Check class, to make sure Checks are not inheriting any messages, and completely own them and have them as fields.

If logging of violation is complicated to move to exact Check, we should make constructor for abstract class to let define violation keys for abstract class. Example: AbstractJavadocCheck.

Goal is to let exact Check module to declare violation keys, to ease catching it by metadata parser and xdoc generation. Such Parsers( that do not use reflection, but use AST) just take all fields by annotation.

I keep this issue in summer project to let me finalize idea later on, and move it out of project.

**Comment by sahitya0xsingh** [comment]
I am on it.

**Comment by romani** [comment]
We did refactoring for this in some Checks, you might find examples already in our code.
Look around for Checks that extend some class.

**Comment by sahitya0xsingh** [comment]
Thanks, @romani ! I’ll look through the existing Checks that extend other classes and use those refactoring patterns as a reference. I’ll start with a few representative cases and then work through the remaining ones.

**Comment by romani** [comment]
Please send PRs for each Check separately, there might be nuances.

**Comment by sahitya0xsingh** [comment]
Sure, thank you! I’ll send a separate PR for each Check and take care of the Check-specific nuances.

**Comment by sahitya0xsingh** [comment]
Hi @romani,

I have researched and audited the codebase to identify all Checks that rely on message constants declared in abstract base classes. I found a total of **21 Checks** across 3 abstract class hierarchies that need to be refactored to declare their own message keys:

### 1. `AbstractSuperCheck` (2 Checks)
- `SuperCloneCheck`
- `SuperFinalizeCheck`

### 2. `AbstractParenPadCheck` (2 Checks)
- `ParenPadCheck`
- `TypecastParenPadCheck`

### 3. `AbstractNameCheck` / `AbstractAccessControlNameCheck` (17 Checks)
- `CatchParameterNameCheck`
- `ConstantNameCheck`
- `LocalVariableNameCheck`
- `TypeNameCheck`
- `MethodNameCheck`
- `MemberNameCheck`
- `ParameterNameCheck`
- `StaticVariableNameCheck`
- `ClassTypeParameterNameCheck`
- `InterfaceTypeParameterNameCheck`
- `MethodTypeParameterNameCheck`
- `RecordTypeParameterNameCheck`
- `RecordComponentNameCheck`
- `LambdaParameterNameCheck`
- `LocalFinalVariableNameCheck`
- `PatternVariableNameCheck`
- `IllegalIdentifierNameCheck`

Should I proceed with creating separate PRs for these?

**Comment by romani** [comment]
Yes, 3 groups, 3 PRs

**Comment by sahitya0xsingh** [comment]
Yes, thank you! I’ll proceed with the 3 groups and create 3 separate PRs accordingly.

**Comment by romani** [comment]
final fix is merged !
@sahitya0xsingh  is our hero!

**Comment by sahitya0xsingh** [comment]
Thank you, @romani! I appreciate your guidance.
